hazelcast/src/main/java/com/hazelcast/internal/jmx/ManagementService.java

```diff
@@ -101,10 +103,17 @@ public void destroy() {
         }
     }
 
-    public static void shutdownAll() {
+    public static void shutdownAll(List<HazelcastInstanceProxy> instances) {
+        for (HazelcastInstanceProxy instance : instances) {
+            shutdown(instance.getName());
```

Comment:
Looks like it already is the case. Beans are registered with the name even it's `null`. (https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/jmx/InstanceMBean.java#L116)
This shoudn't be a problem.